packages/apollo-server/src/__tests__/index.test.ts

```diff
@@ -54,6 +69,65 @@ describe('apollo-server', () => {
       expect(fn.mock.calls).toEqual([['a'], ['b'], ['c'], ['d']]);
     });
 
+    describe('stops even with open HTTP connections', () => {
+      it('all connections are idle', async () => {
+        const server = new ApolloServer({
+          typeDefs,
+          resolvers,
+          // Disable killing non-idle connections. This means the test will only
+          // pass if the fast graceful close of the idle connection works.
+          stopGracePeriodMillis: Infinity,
+        });
+        const { address, port } = await server.listen({ port: 0 });
+
+        // Open a TCP connection to the server, and let it dangle idle
+        // without starting a request.
+        const connectionBarrier = new Barrier();
+        createConnection({ host: address, port: port as number }, () =>
+          connectionBarrier.unblock(),
+        );
+        await connectionBarrier.wait();
+
+        // Stop the server. Before, when this was just net.Server.close, this
+        // would hang. Now that we use stoppable, the idle connection is immediately
+        // killed.
+        await server.stop();
```

Comment:
No `expect`s needed since the only expectation is for this to finish, right?

Reply:
Correct.  Both tests fail with Jest timeout if you revert back to calling httpServer.close instead of httpServer.stop.